### PR TITLE
Vntuananhbui multimodal rag guardrail

### PR DESCRIPTION
**Pull request by vntuananhbui**
# Description

This pull request introduces guardrail functionality into the LlamaIndex Multimodal RAG workflow, leveraging **LLM Guard** to enhance the safety of both input queries and output responses. The following changes are made:
- Integrated `InputScanner` and `OutputScanner` to detect risks such as toxicity, sensitive content, and compliance issues.
- Modified the `CustomQueryEngine` to incorporate guardrails in the query processing pipeline.
- Added a detailed workflow diagram and updated the `README.md` with examples of scanner usage and configuration.

### Motivation
The addition of guardrails ensures that user interactions with LlamaIndex remain safe and compliant with ethical, contextual, security, and compliance standards. This contribution aligns with the growing need for responsible AI usage in applications such as RAG workflows.

### Dependencies
- **LLM Guard** is required to implement input and output scanners.
- No additional dependencies beyond those already specified in `pyproject.toml`.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

To ensure the new functionality works as intended:
- Added unit tests to validate the behavior of `InputScanner` and `OutputScanner`.
- Tested the `CustomQueryEngine` with various queries to ensure it properly blocks unsafe input/output.
- Verified the updated documentation and examples in the `README.md`.

## Suggested Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I ran `make format; make lint` to appease the lint gods.

### CODE REVIEW

File: docs/docs/examples/multimodal-rag-guardrail-gemini/multimodal_rag_guardrail_gemini_llmguard.ipynb
Comment (logan-markewich) [inline docs/docs/examples/multimodal-rag-guardrail-gemini/multimodal_rag_guardrail_gemini_llmguard.ipynb:1]: This should probably mention in the header that this is a service offered through LLM-Guard?

File: docs/docs/examples/multimodal-rag-guardrail-gemini/README.md
Comment (logan-markewich) [inline docs/docs/examples/multimodal-rag-guardrail-gemini/README.md:1]: the examples folder here is meant for notebooks. No need to include a readme

File: docs/docs/examples/multimodal-rag-guardrail-gemini/multimodal_rag_guardrail_gemini_llmguard.ipynb
Comment (vntuananhbui) [reply docs/docs/examples/multimodal-rag-guardrail-gemini/multimodal_rag_guardrail_gemini_llmguard.ipynb:1]: [Fix] I've edited the header to "Multimodal RAG Pipeline with Guardrails Provided by LLM-GUARD" as well as added information that users can extend guardrail from other frameworks.

File: docs/docs/examples/multimodal-rag-guardrail-gemini/README.md
Comment (vntuananhbui) [reply docs/docs/examples/multimodal-rag-guardrail-gemini/README.md:1]: [FIX] I deleted the readme file and image file.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**